src/redfishpower/redfishpower.c

```diff
@@ -314,6 +327,59 @@ static void powermsg_init_curl(struct powermsg *pm)
         Curl_easy_setopt((pm->eh, CURLOPT_HTTPGET, 1));
 }
 
+static char *resolve_hosts_url(const char *hostname, const char *path)
+{
+    char *url;
+    char ipstr[INET6_ADDRSTRLEN] = {0};
+    size_t len;
+    struct addrinfo *ai;
+    struct addrinfo *res = NULL;
+    int ret;
+
+    if ((ret = getaddrinfo(hostname, NULL, NULL, &res)))
+        err_exit(false, "getaddrinfo: %s", gai_strerror (ret));
```

Comment:
I think you need to free `ret` with `freeaddrinfo()` on the exit paths.

Also, maybe try running tests under valgrind to see if anything pops up?

Reply:
ahhh good catch!